activerecord/test/cases/adapters/mysql2/schema_test.rb

```diff
@@ -67,11 +67,11 @@ def test_primary_key
 
       def test_table_exists?
         name = @omgpost.table_name
-        assert @connection.table_exists?(name), "#{name} table should exist"
+        ActiveSupport::Deprecation.silence { assert @connection.table_exists?(name), "#{name} table should exist" }
```

Comment:
let's rewrite this test to use `data_source_exists?`. Make sure to also change the name.